### ISSUE TITLE
Model not fitting on less training data.

### ISSUE THREAD

**Issue by SaiAdityaG** [rbgirshick/py-faster-rcnn#176]
I have trained successfully on PASCAL VOC train dataset. Then I have created two subsets consisting of 10( of 20)  classes each. First subset consists of classes containing more number of images (person, car ,dog, etc) while second subset consists of classes containing less number of images( boat, cow sofa, etc). Training happens successfully on both of sets without any errors. First subset is giving reasonable results but  for the second subset the network classifies every region proposal as background class with high confidence score ( >0.85) where as for other classes it gives small fixed scores (in range of 0.02 - 0.05).
I have tried changing the learning rate, number of iterations, model (VGG and ZF) etc but outcome remains the same.

**Comment by happyharrycn** [comment]
This is a typical problem of a "deep" neural network. You won't be able to train a meaning ful model without sufficient training samples. You can try to (1) remove the dropout layer to facilitate model fitting; (2) freeze the RPN layers (or even fully connected layers) to reduce the number of parameters in training.

**Comment by SaiAdityaG** [comment]
I was training using 'end-to-end' method and there were no detections on test set.  Model converged when I tried with 'alternating training' method.
